### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by Feliverse**
## Description
In this project, you will create a form of UI for your school library app. This way it can be invoked as an executable and not something you use in IRB exclusively.

### Project requirements
- Watch the video with the UI example again (you have seen it while reading the Sneak Peek).
    - [UI example](https://www.youtube.com/watch?v=vkkgrhD6aXQ)
- Your console app should behave in the same way as it is in the example.
- Create a `app.rb` file that will serve as your console app entry-point. It should have methods that do the following:
  - List all books.
  - List all people.
  - Create a person (teacher or student, not a plain `Person`).
  - Create a book.
  - Create a rental.
  - List all rentals for a given person id.
- In your `main.rb` define the entry point, this will be a method called `main` that is invoked at the end of your file. This method should do the following:
  - Present the user with a list of options to perform.
  - Lets users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.

### Need a big picture?

Remind me about [the big picture of this project](./sneak_peek.md).

### CODE REVIEW

**Review by codecaiine — COMMENTED** [review]
Hi @Feliverse , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter errors. Please make sure to fix the errors and all your linters check green 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    def list_all_books(books)
+      if books.empty?
+        puts "\n NO BOOKS REGISTERED!"
+      else
+        puts "\n"
+        books.each do |book|
+          puts "Title: #{book.title}"
+          puts "Author #{book.author}"
+          puts '========================'
+          puts
```
Comment (AmaduKamara) [inline app.rb]: - [x] After creating a book, and trying to list all books, the app crashes with errors shown in the screenshot below in displaying the book's author. Kindly fix that 👍🏽

Screenshot 2023-04-20 at 10 37 20 PM

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    def create_rental(rentals, books, peoples)
+      puts "\n"
+      if @books.empty?
+        puts "\nNO BOOKS TO RENT!\n"
+      elsif @peoples.empty?
+        puts "\nNO PEOPLE TO RENT!\n"
+      else
+        puts 'Select a Book: '
+        @books.each_with_index { |b, i| puts "#{i}: #{b.title} (#{b.author}) " }
+        print 'Book Index: '
```
Comment (AmaduKamara) [inline app.rb:127]: - [x] The app crashes when I try to create a rental for a book. It shows the error below in the screenshot relating book's author. Please work on that 👍🏽

Screenshot 2023-04-20 at 10 43 19 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES ♻️ ♻️ ♻️

Hi @Feliverse,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [x] Kindly work on the errors in displaying a book's author and be sure that everything works fine, all books are displayed when created 👍🏽
- [x] Kindly work on the functionality for creating a rental as it's crashing the app at the moment. 👍🏽
- [x] Please check all features and be sure everything works fine before asking for a review 👍🏽 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app.rb (hunk 1 above)
Comment (Feliverse) [reply app.rb]: Ok, thanks, I will fix

File: app.rb (hunk 2 above)
Comment (Feliverse) [reply app.rb:127]: similar issue, with an error in the author, I will fix

**Review by Reem-lab — APPROVED** [review]
# Hi @Feliverse,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯


## Highlights 🔥🔥🔥🔥
- Followed Gitflow ✅
- Great working on the functionality for UI🔥🔥
- Descriptive PR 🤩✅




## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**